From 53c3f29e67660edcd2c3b5f6a468f361275d17fd Mon Sep 17 00:00:00 2001
From: Fabian Vogt <fabian@ritter-vogt.de>
Date: Tue, 13 Jun 2017 21:04:27 +0200
Subject: [PATCH] Revert "Make WindowSwitching (Alt+Tab) the default left touch
 screen edge"

This reverts commit c453eb696cc97208f8ebf2f632995b891bb425c3.

This causes a 1px wide window on the left screen edge
that steals clicks from windows. Very annoying.

---
 kcmkwin/kwinscreenedges/touch.cpp | 4 +---
 tabbox/tabbox.cpp                 | 6 +++---
 2 files changed, 4 insertions(+), 6 deletions(-)

Index: kwin-5.15.80git.20191114T141940~cb7022150/kcmkwin/kwinscreenedges/touch.cpp
===================================================================
--- kwin-5.15.80git.20191114T141940~cb7022150.orig/kcmkwin/kwinscreenedges/touch.cpp	2019-11-14 15:19:40.000000000 +0100
+++ kwin-5.15.80git.20191114T141940~cb7022150/kcmkwin/kwinscreenedges/touch.cpp	2019-11-14 23:02:35.305453854 +0100
@@ -251,7 +251,7 @@
     KConfigGroup tabBoxConfig(m_config, "TabBox");
     list.clear();
     // TabBox
-    list.append(int(ElectricLeft));
+    list.append(int(ElectricNone));
     list = tabBoxConfig.readEntry("TouchBorderActivate", list);
     foreach (int i, list) {
         monitorChangeEdge(ElectricBorder(i), int(TabBox));
@@ -346,8 +346,6 @@
     // Clear all edges
     for (int i = 0; i < 8; i++)
         m_ui->monitor->selectEdgeItem(i, 0);
-    // select TabBox
-    m_ui->monitor->selectEdgeItem(int(Monitor::Left), int(TabBox));
 }
 
 void KWinScreenEdgesConfig::monitorShowEvent()
Index: kwin-5.15.80git.20191114T141940~cb7022150/tabbox/tabbox.cpp
===================================================================
--- kwin-5.15.80git.20191114T141940~cb7022150.orig/tabbox/tabbox.cpp	2019-11-14 15:19:40.000000000 +0100
+++ kwin-5.15.80git.20191114T141940~cb7022150/tabbox/tabbox.cpp	2019-11-14 23:02:35.305453854 +0100
@@ -782,14 +782,14 @@
         borderConfig = QStringLiteral("BorderAlternativeActivate");
     }
 
-    auto touchConfig = [this, config] (const QString &key, QHash<ElectricBorder, QAction *> &actions, TabBoxMode mode, const QStringList &defaults = QStringList{}) {
+    auto touchConfig = [this, config] (const QString &key, QHash<ElectricBorder, QAction *> &actions, TabBoxMode mode) {
         // fist erase old config
         for (auto it = actions.begin(); it != actions.end(); ) {
             delete it.value();
             it = actions.erase(it);
         }
         // now new config
-        const QStringList list = config.readEntry(key, defaults);
+        const QStringList list = config.readEntry(key, QStringList());
         for (const auto &s : list) {
             bool ok;
             const int i = s.toInt(&ok);
@@ -802,7 +802,7 @@
             actions.insert(ElectricBorder(i), a);
         }
     };
-    touchConfig(QStringLiteral("TouchBorderActivate"), m_touchActivate, TabBoxWindowsMode, QStringList{QString::number(int(ElectricLeft))});
+    touchConfig(QStringLiteral("TouchBorderActivate"), m_touchActivate, TabBoxWindowsMode);
     touchConfig(QStringLiteral("TouchBorderAlternativeActivate"), m_touchAlternativeActivate, TabBoxWindowsAlternativeMode);
 }
 
